optional_plugins/resultsdb/avocado_resultsdb/__init__.py

```diff
@@ -0,0 +1,172 @@
+from avocado.core.settings import settings
+from avocado.core import exceptions
+from avocado.utils import stacktrace
+
+
+class ResultsdbResult(ResultEvents):
+
+    name = 'resultsdb'
+    description = 'Resultsdb result support'
+
+    def __init__(self, args):
+
+        self.rdbapi = None
+        if getattr(args, 'resultsdb_api', None) is not None:
+            self.rdbapi = resultsdb_api.ResultsDBapi(args.resultsdb_api)
+
+        self.rdblogs = None
+        if getattr(args, 'resultsdb_logs', None) is not None:
+            self.rdblogs = args.resultsdb_logs
+
+        self.job_id = None
+        self.job_logdir = None
+
+    def pre_tests(self, job):
+        if self.rdbapi is None:
+            return
+
+        self.job_id = job.unique_id
+        self.job_logdir = os.path.basename(job.logdir)
+
+        ref_url = None
+        if self.rdblogs is not None:
+            ref_url = '%s/%s' % (self.rdblogs, self.job_logdir)
+
+        self.rdbapi.create_group(self.job_id, ref_url)
+
+    def start_test(self, result, state):
+        pass
+
+    def end_test(self, result, state):
+        if self.rdbapi is None:
+            return
+
+        status = state['status']
+        outcome = self._status_map(status)
+
+        name = state['name'].name
+
+        group = [self.job_id]
+
+        note = None
+        if state['fail_reason'] is not None:
+            note = str(state['fail_reason'])
+
+        ref_url = None
+        if self.rdblogs is not None:
+            logdir = os.path.basename(state['logdir'])
+            ref_url = '%s/%s/test-results/%s' % (self.rdblogs,
+                                                 self.job_logdir,
+                                                 logdir)
+
+        local_time_start = time.localtime(state['time_start'])
+        local_time_end = time.localtime(state['time_end'])
+        data = {'time_elapsed': "%.2f s" % state['time_elapsed'],
+                'time_start': time.strftime("%Y-%m-%d %H:%M:%S",
+                                            local_time_start),
+                'time_end': time.strftime("%Y-%m-%d %H:%M:%S",
+                                          local_time_end),
+                'logdir': state['logdir'],
+                'logfile': state['logfile'],
+                'whiteboard': state['whiteboard'],
+                'status': status}
+
+        self.rdbapi.create_result(outcome,
+                                  name,
+                                  group,
+                                  note,
+                                  ref_url,
+                                  **data)
+
+    def test_progress(self, progress=False):
+        pass
+
+    def post_tests(self, job):
+        pass
+
+    @staticmethod
+    def _status_map(status):
+        """
+        Returns the resultsdb corresponding status to Avocado status.
+        Valid statuses are:
+         PASSED
+         FAILED
+         INFO
+         NEEDS_INSPECTION
```

Comment:
This is hideously long, how about putting it into a single line? `PASSED/FAILED/INFO/NEEDSS_INTROSPECTION`. Also it might be useful to mention on several places that this is based on resultsdb-2.0 as resultsdb-1.0 supported other statuses and the future versions might also be more-less incompatible...